### PR TITLE
bugfix collector can not auto reconnect when channel idle

### PR DESCRIPTION
**Pull request by tomsun28**
## What's changed?

bugfix collector can not auto reconnect when channel idle

## Checklist

- [x]  I hereby agree to the terms of the [HertzBeat CLA](https://gist.github.com/tomsun28/511c04e7643901cb550bb6ecc75a661b)
- [ ]  I have read the [Contributing Guide](https://hertzbeat.com/docs/others/contributing/)
- [ ]  I have written the necessary doc or comment.
- [ ]  I have added the necessary unit tests and all cases have passed.

### CODE REVIEW

File: collector/src/main/java/org/dromara/hertzbeat/collector/dispatch/entrance/CollectServer.java [hunk 1]
```diff
@@ -132,12 +132,16 @@ public void onChannelActive(Channel channel) {
...
-                    ClusterMsg.Message heartbeat = ClusterMsg.Message.newBuilder()
-                            .setIdentity(identity)
-                            .setType(ClusterMsg.MessageType.HEARTBEAT)
-                            .build();
-                    CollectServer.this.sendMsg(heartbeat);
-                    log.info("collector send cluster server heartbeat, time: {}.", System.currentTimeMillis());
+                    try {
+                        ClusterMsg.Message heartbeat = ClusterMsg.Message.newBuilder()
+                                .setIdentity(identity)
+                                .setType(ClusterMsg.MessageType.HEARTBEAT)
```
Comment (Ceilzcx) [inline collector/src/main/java/org/dromara/hertzbeat/collector/dispatch/entrance/CollectServer.java:138]: 发消息最好把Request加上

File: collector/src/main/java/org/dromara/hertzbeat/collector/dispatch/entrance/CollectServer.java [hunk 2]
```diff
@@ -132,12 +132,16 @@ public void onChannelActive(Channel channel) {
...
+                                .build();
+                        CollectServer.this.sendMsg(heartbeat);
+                        log.info("collector send cluster server heartbeat, time: {}.", System.currentTimeMillis());   
+                    } catch (Exception e) {
+                        log.error(e.getMessage());
```
Comment (Ceilzcx) [inline collector/src/main/java/org/dromara/hertzbeat/collector/dispatch/entrance/CollectServer.java:143]: 这里log日志弄的详细一点把

File: manager/src/main/java/org/dromara/hertzbeat/manager/scheduler/netty/ManageServer.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ private void init(final SchedulerProperties schedulerProperties, final CommonThr
         this.remotingServer.registerHook(Lists.newArrayList(new NettyHook() {
             @Override
             public void doBeforeRequest(ChannelHandlerContext ctx, ClusterMsg.Message message) {
-                ManageServer.this.clientChannelTable.put(message.getIdentity(), ctx.channel());
+                // do something before processor list
```
Comment (Ceilzcx) [inline manager/src/main/java/org/dromara/hertzbeat/manager/scheduler/netty/ManageServer.java:69]: 去了的话直接删了好了，hook不用加了

File: remoting/src/main/java/org/dromara/hertzbeat/remoting/netty/NettyRemotingServer.java [hunk 4]
```diff
@@ -138,7 +138,7 @@ private void initChannel(final SocketChannel channel) {
         pipeline.addLast(new ProtobufVarint32LengthFieldPrepender());
         pipeline.addLast(new ProtobufEncoder());
         // idle state
-        pipeline.addLast(new IdleStateHandler(0, 0, 30));
+        pipeline.addLast(new IdleStateHandler(0, 0, 100));
```
Comment (Ceilzcx) [inline remoting/src/main/java/org/dromara/hertzbeat/remoting/netty/NettyRemotingServer.java:141]: 放到ServerConfig里面怎么样？

File: collector/src/main/java/org/dromara/hertzbeat/collector/dispatch/entrance/CollectServer.java (hunk 1 above)
Comment (tomsun28) [reply collector/src/main/java/org/dromara/hertzbeat/collector/dispatch/entrance/CollectServer.java:138]: 好的 `setDirection(ClusterMsg.Direction.REQUEST)`

File: manager/src/main/java/org/dromara/hertzbeat/manager/scheduler/netty/ManageServer.java (hunk 3 above)
Comment (tomsun28) [reply manager/src/main/java/org/dromara/hertzbeat/manager/scheduler/netty/ManageServer.java:69]: 好 那接口还是留着以后用

File: remoting/src/main/java/org/dromara/hertzbeat/remoting/netty/NettyRemotingServer.java (hunk 4 above)
Comment (tomsun28) [reply remoting/src/main/java/org/dromara/hertzbeat/remoting/netty/NettyRemotingServer.java:141]: +1

**Review by Ceilzcx — APPROVED** [review]

**Pull request merged**